### ISSUE TITLE
Invalid redirect URI

### ISSUE THREAD

**Issue by Breakingx** [matt8707/ha-fusion#356]
![IMG_6710](https://github.com/matt8707/ha-fusion/assets/19574599/35058ace-82e3-4b13-8d3c-735a505067f9)

Hey,

I‘m currently using nabu Casa and everything I try to open fusion on a mobile device this screen appears. On my laptop it’s working fine. 

In the logs isn’t anything meaningful. 
this occurs even on local access.

**Comment by thundergreen** [comment]
having same issue ....

**Comment by Youri3246** [comment]
Same here...

**Comment by dadalouis** [comment]
me 2

**Comment by onidaR** [comment]
Same here. Can someone help with this pls? 
Thanks.

**Comment by su2626** [comment]
Same here too.
Worked like a charm from app (localy and remotely) until I switched user to do some tests. Since then, Invalid redirect URI

**Comment by dennidoubles** [comment]
Same here...it won't start on mobile.
`Invalid redirect URI`

**Comment by imMrRAW** [comment]
Same issue

**Comment by huseyinulas** [comment]
same here

**Comment by anihao** [comment]
same here

**Comment by vladimirssa** [comment]
Same here

**Comment by zapatos-design** [comment]
having a similar problem. worked great for a while using HA Companion, then got the URI error and now nothing. Only way to access is from local browser.

**Comment by bkhalterman** [comment]
Same problem here.

**Comment by Ratty7198** [comment]
I have found a workaround. If you use wallpanel app everything seems fine Web Link: https://wallpanel.xyz/

**Comment by bkhalterman** [comment]
> I have found a workaround. If you use wallpanel app everything seems fine Web Link: https://wallpanel.xyz/

How can I edit the fusion configuration to enable wallpanel? I want to use wallpanel to hide the sidebar and top bar.

**Comment by Ratty7198** [comment]
> > I have found a workaround. If you use wallpanel app everything seems fine Web Link: https://wallpanel.xyz/
> 
> How can I edit the fusion configuration to enable wallpanel? I want to use wallpanel to hide the sidebar and top bar.

I got the page address from a browser page and input that as the page to display in wallpanel as for hiding the sidebar you should be able to do that from the usual place in the hamburger menu.

**Comment by bkhalterman** [comment]
> > > I have found a workaround. If you use wallpanel app everything seems fine Web Link: https://wallpanel.xyz/
> > 
> > 
> > How can I edit the fusion configuration to enable wallpanel? I want to use wallpanel to hide the sidebar and top bar.
> 
> I got the page address from a browser page and input that as the page to display in wallpanel as for hiding the sidebar you should be able to do that from the usual place in the hamburger menu.

Yes, I was able to get wallpanel working on my tablet, no issue with the URL. When I hide the "sidebar" in homeassistant user settings, it then creates a top bar. Seems like I can't get away from having something interfering with my dashboard. So I was looking into the wallpanel HACS integration as a potential solution to removing the sidebar. I know it works with normal dashboards, but was unable to figure out how to integrate it with ha-fusion.

**Comment by sShrek1** [comment]
I did some small debugging. I am guessing you get this url in HA companion app. 
Seems that it is fine until it redirects to the HTTP local instance for authentication. 

If anyone have time you can enable WebView debug in HA companion app, connect to PC (chrome?) and inspect the redirect URL.

**Comment by socekin** [comment]
same here

**Comment by jankcy** [comment]
same here，PC chrome work well。I don't know what url the app error mean？

**Comment by Harti33** [comment]
Still got this problem!

**Comment by Itsememario** [comment]
It Seems like homeassistants New ingress Update Broke it

**Comment by Figurenbond** [comment]
![IMG_B5266B3C29A7-1](https://github.com/matt8707/ha-fusion/assets/131239928/62226fe7-fa78-42a2-ad04-67defeee21d2)
 Same issue

**Comment by 2wenty2wo** [comment]
Same for me on iPhone

**Comment by andrefernandes86** [comment]
Same issue here! I am using Nabu Casa.

**Comment by sami-juvonen** [comment]
Same

**Comment by lccox** [comment]
In one of the closed issues, there is a solution regarding "Invalid redirect URI". It's necessary to change the port in the settings of the ha-fusion add-on, and directly access port 8099.

**Comment by PepperHead25** [comment]
Sorry if this is a dumb question as I'm new to all of this.  I can't get Fusion to load on my Andriod phone app.  I've gone into the addon under config and it seems to already be set to 8099.  I'm not however on a static IP if that makes a difference.  I would really like to be able to use this dashboard on my cell phone.   Any suggestions would be appreciated.

**Comment by thomasv04** [comment]
Same here, and i have a OVH domain and cloudflare

**Comment by matt8707** [comment]
There's an issue specifically with the combination of the ingress and companion app. I've implemented support for long-lived access tokens to address this scenario.

**Comment by Gerrett84** [comment]
2024.4.0 fixed it 👍🏻

**Comment by MichalStrzelczyk** [comment]
2024.7.1 the same issue

**Comment by FrankTub** [comment]
I have the same issue. I tried to change the port in the settings to `8099` and now it shows me that it is probably an interaction between using Cloudflare and this app. Pointing it to port `8099` changes the error message I get to a 502:

<img width="502" alt="image" src="https://github.com/user-attachments/assets/98c8e7b1-0577-4d4a-bb76-a1ea43fb4ae8">

I'm on `2024.10.1`.

**Comment by FrankTub** [comment]
Perhaps this helps someone here as well. For me what worked to resolve this, is clear the cache in CloudFlare and retry. Then it just pops up as expected (I guess?), since then you have to create a long lived token and get a visible UI.
